### PR TITLE
Add preliminary support for doubles.

### PR DESCRIPTION
**Pull request by johanbrandhorst**
Does currently not include any way to handle floating point inaccuracies.

I thought I'd get the ball rolling on #11. Obviously this does not support an "epsilon" for handling floating point inaccuracies. Please provide feedback. How would you like the epsilon comparison to be implemented?

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
Oh yeah and I changed a few variable names that I thought were perhaps slightly inaccurate, (i.e. "higherThan" -> "greaterThan"). Also the comment on the proto file from "strict" to "strictly". Sounds better to me, feedback welcome :).

**Comment by johanbrandhorst** [comment]
Alright I added epsilon support now, is it how you imagined it?

File: plugin/plugin.go [hunk 1]
```diff
@@ -129,6 +129,14 @@ func (p *plugin) isSupportedInt(field *descriptor.FieldDescriptorProto) bool {
 	return false
 }
 
+func (p *plugin) isSupportedDouble(field *descriptor.FieldDescriptorProto) bool {
+	switch *(field.Type) {
+	case descriptor.FieldDescriptorProto_TYPE_DOUBLE:
```
Comment (mwitkow) [inline plugin/plugin.go]: I think the golang codegen will work for floats, maybe worth adding? `FieldDescriptorProto_TYPE_FLOAT` here?

**Comment by johanbrandhorst** [comment]
Ah, I didn't even know there was a float type in the proto definition. Float will map to golang type *float32 though, want me to add both floats and doubles to cover the cases?

**Comment by johanbrandhorst** [comment]
It appears gogoproto does not like it when I define float_gt and float_lt using the proto "float" type, but it works with the "double" type. Would you like to keep the types the same and just keep the variable names and methods called float?

**Comment by johanbrandhorst** [comment]
Let me know if you have anymore comments

**Comment by mwitkow** [comment]
This is great, thank you! :)

**Pull request merged**